src/CachePlugin.php

```diff
@@ -36,11 +42,13 @@ class CachePlugin implements Plugin
 
     /**
      * @param CacheItemPoolInterface $pool
+     * @param StreamFactory          $streamFactory
      * @param array                  $options
      */
-    public function __construct(CacheItemPoolInterface $pool, array $options = [])
+    public function __construct(CacheItemPoolInterface $pool, StreamFactory $streamFactory, array $options = [])
```

Comment:
I think this is a great use case why options should probably moved out of the constructor and configured by a setter.